packages/widget/src/components/amount-selector/amount-selector.ts

```diff
@@ -137,6 +137,12 @@ export class AmountSelector extends BaseComponent {
     );
   }
 
+  updated(changedProperties: PropertyValues): void {
+    if (changedProperties.has('resources')) {
```

Comment:
shouldn't this look for change in selectedResource?

Reply:
if I change network, the resource is not the same, and whenever we change networks the resources are updated

Reply:
But selected resource should update to `undefined` if previously selected resource is not on the updated resource list^^

Reply:
seeking clarification here:

- expected behaviour is if you change source but on your dest network remains the same and the route is valid, like:

`from: cronos => mumbai`
`to: sepolia`

In this case, should we reset the balance to 0 and then you select the resource to fetch the balance?

- what happens when you want switch to an invalid route, like from sepolia/mumbai to cronos/mumbai. Currently with the changes here, we are reseting the destination dropdown since that route doesn't exist, should we reset the balance to zero?

I'm asking this because as user I would expect, since there is no token selected, that if I change the network and the token gets de-selected, the balance goes to 0 😄

Reply:
- if you are changing source network to supported route
   - reset balance to zero
   - reset selected token if doesn't exist on the list of resources
- if you are changing source network to unsupported route
   - reset balance and selected token and destination network (network selector component should handle that because selected dest network is not on list of networks)

Reply:
@mpetrunic I think the flow is working as expected, it took me some time to solve this "detail". Let me know if you have more feedback